specs/psl-sql-conversion.md

```diff
@@ -0,0 +1,459 @@
+  "viewCount" integer NOT NULL DEFAULT 0 ,
+  PRIMARY KEY ("id")
+);
+```
+
+### Mapping a model with scalar list fields
+
+Prisma Schema Example:
+```groovy
+model User {
+    id Int @id
+    ints Int[]
+}
+```
+
+A scalar list field in PSL is mapped to a separate table for the values. The name follows the `Model_fieldname` convention. The important columns contained are nodeId for the specific node a value is associated with and value for the actual value. Position is generated and should allow fancy operations like ordered inserts, but is not used. 
+
+Resulting SQL Example (Postgres):
+```sql
+CREATE TABLE "test"."User" (
+  "id" SERIAL,
+  PRIMARY KEY ("id")
+);
+
+CREATE TABLE "test"."User_ints" (
+  "nodeId" integer   REFERENCES "test"."User"("id") ON DELETE CASCADE,
+  "position" integer NOT NULL  ,
+  "value" integer NOT NULL  ,
+  PRIMARY KEY ("nodeId","position")
+);
+```
+
+### Mapping a model with a unique field
+
+```groovy
+model Blog {
+    id        String @id @default(cuid())
+    title     String
+    slug      String @unique
+}
+```
+
+For a field that is annotated with `@unique` a unique index is created in the underlying database.
+
+Resulting SQL Example (Postgres):
+```sql
+CREATE TABLE "test"."Blog" (
+  "id" text NOT NULL  ,
+  "slug" text NOT NULL DEFAULT '' ,
+  "title" text NOT NULL DEFAULT '' ,
+  PRIMARY KEY ("id")
+);
+
+CREATE UNIQUE INDEX "Blog.slug" ON "test"."Blog"("slug")
+```
+
+### Mapping a model with a compound unique field
+
+```groovy
+model User {
+    id Int @id
+    firstname String
+    lastname String
+    @@unique([firstname, lastname], name: "test")
+}
+```
+
+For the block attribute `@@unique` a unique index is created in the underlying database.
+
+Resulting SQL Example (Postgres):
+```sql
+CREATE TABLE "test"."User" (
+  "firstname" text NOT NULL DEFAULT '' ,
+  "id" SERIAL,
+  "lastname" text NOT NULL DEFAULT '' ,
+  PRIMARY KEY ("id")
+);
+
+CREATE UNIQUE INDEX "test" ON "test"."User"("firstname","lastname")
+```
+
+
+## Mapping Enums
+
+```groovy
+model User {
+    id Int @id
+    enum TestEnum
+}
+
+enum TestEnum{
+  A
+  B
+}
+```
+
+We handle the validation of Enum values in the core, and place no constraints on the db column.  
+
+```sql
+CREATE TABLE "test"."User" (
+  "enum" text NOT NULL DEFAULT 'A' ,
+  "id" SERIAL,
+  PRIMARY KEY ("id")
+);
+```
+
+🚨 We do not use any DB capabilities for enums, therefore when introspecting again, this is just a String field for us. 
+
+
+
+## Mapping Default Values 
+
+```groovy
+model User {
+  id       Int       @id
+  string   String    @default("test")
+  date     DateTime  @default(now())
+}
+```
+
+```sql
+CREATE TABLE "test"."User" (
+  "date" timestamp(3) NOT NULL DEFAULT '1970-01-01 00:00:00' ,
+  "id" SERIAL,
+  "string" text NOT NULL DEFAULT 'test' ,
+  PRIMARY KEY ("id")
+);
+```
+
+🚨 DateTime default values don't yet work. The `now()` expression should translate to something like `CURRENT_TIMESTAMP`. 
```

Comment:
I'm not sure what the migration engine does with this now, should I check?

Reply:
I assume it defaults to 1970, but please check.